Modules/EMCAL/src/PayloadPerEventDDLCheck.cxx

```diff
@@ -0,0 +1,109 @@
+#include "QualityControl/MonitorObject.h"
+#include "EMCAL/PayloadPerEventDDLCheck.h"
+#include "QualityControl/QcInfoLogger.h"
+#include "QualityControl/Quality.h"
+
+// ROOT
+#include <TH1.h>
+#include <TH2.h>
+#include <TPaveText.h>
+#include <TLatex.h>
+#include <TList.h>
+#include <TRobustEstimator.h>
+#include <ROOT/TSeq.hxx>
+#include <iostream>
+#include <vector>
+#include <TCanvas.h>
+#include <string>
+
+using namespace std;
+
+namespace o2::quality_control_modules::emcal
+{
+// configure threshold-based checkers
+auto nChiSqMedPayloadThresh = mCustomParameters.find("ChiSqMedPayloadThresh");
+if (nChiSqMedPayloadThresh != mCustomParameters.end()) {
+  try {
+    mChiSqMedPayloadThresh = std::stod(nChiSqMedPayloadThresh->second);
+  } catch (std::exception& e) {
+    ILOG(Error, Support) << ("Value {} not a double", nChiSqMedPayloadThresh->second.data()) << ENDM;
+  }
+}
+
+auto nBadChiSqLowPayloadThresh = mCustomParameters.find("BadChiSqLowPayloadThresh");
+if (nBadChiSqLowPayloadThresh != mCustomParameters.end()) {
+  try {
+    mChiSqBadLowPayloadThresh = std::stod(nBadChiSqLowPayloadThresh->second);
+  } catch (std::exception& e) {
+    ILOG(Error, Support) << ("Value {} not a double", nBadChiSqLowPayloadThresh->second.data()) << ENDM;
+  }
+}
+}
+
+Quality PayloadPerEventDDLCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+  if (!mCalibReference) {
+    // Load reference histogram from the CCDB
+    mCalibReference = this->retrieveConditionAny<TH2>("/EMC/QCREF/PayloadSizeCheck");
+  }
+
+  auto mo = moMap->begin()->second;
+  Quality result = Quality::Good;
+
+  if (mo->getName() == "PayloadSizePerDDL") {
+
+    auto* h = dynamic_cast<TH2*>(mo->getObject());
+
+    for (int i = 1; i <= h->GetNbinsX(); i++) {
+      std::unique_ptr<TH1> h_y = (TH1*)h->ProjectionY(Form("h_y_%d", i), i, i);
```

Comment:
Also this is not valid code, the operator= to pointers is not supported. The line must be changed to 
```c++
std::unique_ptr<TH1> h_y(h->ProjectionY(Form("h_y_%d", i), i, i));
```
To initialise a smart pointer this has to be done in the constructor or by setter functions, otherwise it cannot do the memory management.